tool/modules/uds_fuzz.py

```diff
@@ -0,0 +1,473 @@
+    reset_type = args.reset
+    session_type = args.sess_type
+    target = args.target_seed
+    reset_delay = args.delay
+    loop = True
+
+    seed_list = []
+    try:
+        print("Security seed dump started. Press Ctrl+C to stop.\n")
+        while loop:
+            # Extended diagnostics
+            ecu_reset(arb_id_request, arb_id_response, reset_type, None)
+            time.sleep(reset_delay)
+            for i in range(0, len(session_type), 4):
+
+                if session_type[i] == "1" and session_type[i+1] == "0":
+                    session = str_to_hex(i, session_type)
+                    response = extended_session(arb_id_request,
+                                                arb_id_response,
+                                                session)
+                    if not Iso14229_1.is_positive_response(response):
+                        print("Unable to enter session. Retrying...\n")
+                        break
+
+                elif session_type[i] == "2" and session_type[i+1] == "7":
+                
+                    # Request seed
+                    session = str_to_hex(i, session_type)
+                    response = request_seed(arb_id_request, arb_id_response,
+                                            session, None, None)
+                    if response is None:
+                        print("\nInvalid response")
+                    elif Iso14229_1.is_positive_response(response):
+                        seed_list.append(list_to_hex_str(response[2:]))
+                        print("Seed received: {}\t(Total captured: {}, Delay used: {})"
+                            .format(list_to_hex_str(response[2:]),
+                                    len(seed_list),reset_delay), end="\r")
+
+                        if list_to_hex_str(response[2:]) == list_to_hex_str(str_to_int_list(target)):
+                            print("\n\nTarget seed found with delay: ", reset_delay)
+                            loop = False
+                            break
+
+                        stdout.flush()
+
+                    else:
+                        print_negative_response(response)
+                        break
+                
+                elif session_type[i] == 1 and session_type[i+1] == 1:
+                    ecu_reset(arb_id_request, arb_id_response, reset_type, None)
+                    time.sleep(reset_delay)
+                else:
+                    break
+
+
+            if reset_type:
+                ecu_reset(arb_id_request, arb_id_response, reset_type, None)
+                time.sleep(reset_delay)
+                reset_delay += 0.001
+
+    except KeyboardInterrupt:
+        print("Interrupted by user.")
+    except ValueError as e:
+        print(e)
+        return
+
+    if len(seed_list) > 0:
+        print("\n")
+        print("Security Access Seeds captured:")
+        for seed in seed_list:
+            print(seed)
+
+def str_to_hex(i, session_type):
```

Comment:
This function lacks length checks before accessing indices, which could lead to trouble